Allow an alternate warp file by passing it as an argument

r0b0tbuilder opened the pull request:
It could be useful to specify another warp file (perhaps, a system-wide warp file).  I made a small change to the script so that the first argument passed to the script is the location of the alternate warp file.

jpalardy commented on warp:
```diff
@@ -56,7 +63,9 @@ warp() {
-  warp
+  # pass the first argument to the warp function
+  # so that it can be used as an alternate warp file
+  warp $1
```
this might be better as `"$@"` to pass all arguments to warp.

jpalardy commented on warp:
```diff
@@ -1,8 +1,15 @@
+  else
+    local SOURCE="$HOME/.warp"
+  fi
+
```
how about:

    local SOURCE="${1:-$HOME/.warp}"

jpalardy commented:
good idea, I made some comments ^

r0b0tbuilder commented:
I have pushed two commits that introduce the changes that you suggested.

r0b0tbuilder commented:
Maybe a small description of this feature could be added to the readme file?

jpalardy commented on warp:
```diff
@@ -56,7 +60,8 @@ warp() {
 if [[ $_ == $0 ]]; then
-  warp
+  # pass all arguments to the warp function
+  warp $@
```
You need double-quotes around $@ to propagate double-quotes to the subprocess.

Not necessary for this, but a good practice.

jpalardy commented:
If you can write it up a snippet for the README, that would be great.

r0b0tbuilder commented:
I have pushed a commit that expands the "how" section of the README to include a short description of the feature that I implemented.  I have also fixed the quotes around $@, as you have suggested.

jpalardy commented:
Looking good, thanks for all this :-)

dchapman1988 commented:
:+1:
